rule/src/main/java/com/buschmais/jqassistant/core/rule/impl/reader/IndentHelper.java

```diff
@@ -0,0 +1,124 @@
+package com.buschmais.jqassistant.core.rule.impl.reader;
+
+public class IndentHelper {
+    /**
+     * This class make the String more effective and understandable. by removing the extra spaces,lines and tabs.
+     * even tho , there is some exceptions where if the tabs are not in order the whole class won't work!
+     * NOTE : Check the TextUtilTest.java to see the examples
+     *
+     * @param text
+     * @return edited text without the unused whitespaces
+     */
+    public static String removeIndent(String text) {
+        if (text == null) {
+            return null;
+        }
+        String textWithoutEmptyLines = removeEmptyLines(text);
+        String[] lines = textWithoutEmptyLines.split("\\n");
+        if (text.isBlank()) {
+            return textWithoutEmptyLines;
+        }
+        int indent = getIndent(lines);
+        return removeIndent(lines, indent);
+    }
+
+    /**
+     * Removes the specified indentation from each line in the provided array of lines. If it's needed.
+     * Removes the trailing and leading spaces from each lines
+     *
+     * @param lines  An array of strings representing lines of text
+     * @param indent The number of leading whitespace characters to remove from each line
+     * @return The text with indentation removed
+     */
+
+    private static String removeIndent(String[] lines, int indent) {
+        StringBuilder resultBuilder = new StringBuilder();
+        // We make a loop here to print the text LINE by LINE. it makes it easier to edit and change!
+        for (int i = 0; i < lines.length; i++) {
+            // We replace all trailing whitespaces from each line. to make it simpler!
+            String line = lines[i].replaceAll("\\s+$", "");
+            if (!line.isBlank()) {
+                resultBuilder.append(line.substring(indent));
+            } else {
+                resultBuilder.append(lines[i]);
+            }
+            if (i < lines.length - 1) {
+                resultBuilder.append("\n");
+            }
+        }
+        return resultBuilder.toString();
+
+    }
+
+    /**
+     * In this method we read the text line by line vertically (in columns) . specially to read  spaces or tabs.
```

Comment:
The description is too "noisy". Describe briefly the expexted outcome: the indent of a  text block is determined by identifying the sequence of blank characters that is shared/the same across all lines of the text block.